assets/js/blocks/cart-checkout/checkout/block.js

```diff
@@ -72,6 +70,16 @@ const Block = ( { attributes, isEditor = false, shippingRates = [] } ) => {
 		},
 	};
 
+	const {
+		shippingRatesLoading,
+		shippingAddress: shippingFields,
+		setShippingAddress: setShippingFields,
+	} = useShippingRates(
+		Object.fromEntries(
+			Object.entries( addressFields ).map( ( [ key ] ) => [ key, '' ] )
+		)
```

Comment:
the current shape of `addressFields` is key => object of meta things, we only something like
```js
{
	country: '',
	state: '',
	city: '',
	postcode: '',
	};
```
so we remap the object to this.

Reply:
the goal is to always and watch and send the fields being used in the form, this ensures that if someone adds custom fields in the future, it should also work, in the cart, we don’t have any sort of dynamic fields yet (the fields are hardcoded in https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/4ddcf447be4f9bce7e973a7c91d9da148d1d2f3d/assets/js/base/components/shipping-calculator/address.js#L22

we might export them somewhere and use them here.

there is also the logic of turning an array to an object, this I will move to the hook.

Reply:
fixed in a73d39d14